framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -38,33 +38,54 @@ MediaPlayerImpl::MediaPlayerImpl()
 
 player_result_t MediaPlayerImpl::create()
 {
+	player_result_t ret = PLAYER_ERROR;
+
 	std::unique_lock<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaPlayer create\n");
-	if (mCurState != PLAYER_STATE_NONE) {
-		meddbg("MediaPlayer create fail : wrong state\n");
-		return PLAYER_ERROR;
-	}
 
 	PlayerWorker& mpw = PlayerWorker::getWorker();
 	mpw.startWorker();
-	mpw.getQueue().enQueue(&MediaPlayerImpl::notifySync, shared_from_this());
+
+	mpw.getQueue().enQueue(&MediaPlayerImpl::createPlayer, shared_from_this(), std::ref(ret));
 	mSyncCv.wait(lock);
 
+	if (ret == PLAYER_ERROR) {
+		mpw.stopWorker();
+	}
+
+	return ret;
+}
+
+void MediaPlayerImpl::createPlayer(player_result_t &ret)
+{
+	medvdbg("MediaPlayer Worker : createPlayer\n");
+	if (mCurState != PLAYER_STATE_NONE) {
+		meddbg("MediaPlayer create fail : wrong state\n");
+		goto errout;
```

Comment:
Here's only place where return PLAYER_ERROR, in this case you don't have to put goto.

Reply:
Fixed it